### DISCUSSION TITLE
Problems when convert PDF to png some characteres are converted on black boxes

### DISCUSSION THREAD

**Discussion by cyberfido** [ImageMagick/ImageMagick6#273] [Q&A]
### ImageMagick version

6.9.12-32

### Operating system

Linux

### Operating system, version and so on

Ubuntu 20.04

### Description

I'm running this command
convert -define bmp:format=bmp3 fail_symbol.pdf test.png
then I added a -verbose to get the information on what is imagemagick sending to ghostscript, and this is the result command:

gs -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r72x72' -dPrinted=false -sOutputFile=test.png fail_symbol.pdf

if I execute that, I got a incorrect output:

![Screenshot_138](https://github.com/ImageMagick/ImageMagick/assets/146215213/e02386e0-5974-4241-be81-65315f462e19)

 with some characters like a blackbox, then if I drop the value -dPrinted=false from the script I got the correct results, now my question is how can I configure imagemagick delegates or which parameter do I have to send to ghostscript to avoid -dPrinted=false  on the script

### Steps to Reproduce

run the command

convert -define bmp:format=bmp3 fail_symbol.pdf test.png

with the PDF Attached

### Images

[fail_symbol.pdf](https://github.com/ImageMagick/ImageMagick/files/12740654/fail_symbol.pdf)

**Comment by fmw42** [comment]
I do not think that in your command -define bmp:format=bmp3 will do anything for the PNG output.  Also the image you post is not what I get from running

`convert fail_symbol.pdf -alpha off test.png`

![test-0](https://github.com/ImageMagick/ImageMagick6/assets/13458218/27de3866-28eb-4f2b-aef8-5a123634560c)

What is your Ghostscript version?  I am running IM 6.9.12-96 with GS 9.56.1 on Mac OSX.  I have see issues with GS 10.x and others have had to downgrade to 9.x.x

If you really need to modify the gs command, I believe that you can do that in the delegates.xml file for PS/PSF/EPS/PDF.

**Reply by cyberfido** [reply]
Hi, I'm using IM 6.9.12-32 and gs 9.50, I tried to modify the delegates file, but didn't work

**Reply by fmw42** [reply]
IM 6.9.12-32 is old as is GS 9.50. Perhaps you should upgrade one or both.

**Reply by cyberfido** [reply]
I tried with IM 6.9.7-10 and GS 9.26 and got correct results, I will check with newer versions to check if it works

**Reply by cyberfido** [reply]
Problem solved with IM 6.9.7-10 and GS 10.02.1  Working on Ubuntu 20.0406  Thanks
